### ISSUE TITLE
Think through delay period options

### ISSUE THREAD

**Issue by cwgoes** [cosmos/ics#508]
And how to push to application, secondary queues, etc. etc.

**Comment by ethanfrey** [comment]
I'm curious. Can you expand a bit more? (I assume this is a personal placeholder issue).

This is when you set/change the per-channel delay period and how that change can safely propagate over all other components in this distributed system which is IBC?

**Comment by cwgoes** [comment]
Yes, this is a placeholder issue from a meeting. The basic idea is that it would be preferable to push the delay period choice out to the application level so that applications can consider the contents of a packet (e.g. how much value is being transferred) and their state (e.g. how much value has been transferred recently) in order to determine what delay period should be imposed for whatever safety guarantees are desired.
